src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -263,10 +264,13 @@ public static SignedPackageArchiveMetadata ReadSignedArchiveMetadata(BinaryReade
                     IsPackageSignatureFile = isPackageSignatureFile,
                     HeaderSize = header.GetSizeInBytes(),
                     OffsetToFileHeader = header.RelativeOffsetOfLocalHeader,
-                    Position = header.OffsetFromStart
+                    Position = header.OffsetFromStart,
+                    IndexInHeaders = index
```

Comment:
Why do you need to add this here?

Reply:
`UpdateSignedPackageArchiveMetadata(...)` is responsible for updating `IndexInHeaders`, but I'm [no longer calling it](https://github.com/NuGet/NuGet.Client/pull/2088/files#diff-ebd06908ad02186c27abc513d43b0677L281) with this change, so I have to update the index here.  `UpdateSignedPackageArchiveMetadata(...)` is still used by tests though.

Also note that `UpdateSignedPackageArchiveMetadata(...)` exists in 4.6.x branches, not dev.  I removed the method in https://github.com/NuGet/NuGet.Client/pull/2044.